modules/plugin/src/main/scala/com/fortysevendeg/hood/plugin/SbtHoodPlugin.scala

```diff
@@ -0,0 +1,146 @@
+import com.fortysevendeg.hood.model.{Benchmark, HoodError}
+import sbt.{AutoPlugin, Def, PluginTrigger, Task}
+import io.chrisdavenport.log4cats.Logger
+import io.chrisdavenport.log4cats.slf4j.Slf4jLogger
+import cats.effect.Console.implicits._
+
+object SbtHoodPlugin extends AutoPlugin with SbtHoodDefaultSettings with SbtHoodKeys {
+
+  override def projectSettings: Seq[Def.Setting[_]] = defaultSettings
+  override val trigger: PluginTrigger               = noTrigger
+
+  def compareBenchmarksTask: Def.Initialize[Task[Unit]] = Def.task {
+
+    implicit val logger = Slf4jLogger.getLogger[IO]
+
+    benchmarkTask(
+      previousBenchmarkPath.value,
+      currentBenchmarkPath.value,
+      keyColumnName.value,
+      compareColumnName.value,
+      thresholdColumnName.value,
+      modeColumnName.value,
+      unitsColumnName.value,
+      generalThreshold.value,
+      benchmarkThreshold.value
+    ).leftFlatMap(e =>
+        EitherT.left[List[BenchmarkComparisonResult]](logger.error(s"There was an error: $e")))
+      .value
+      .unsafeRunSync()
+
+    ()
+  }
+
+  def benchmarkTask[F[_]](
+      previousPath: File,
+      currentPath: File,
+      keyColumnName: String,
+      compareColumnName: String,
+      thresholdColumnName: String,
+      modeColumnName: String,
+      unitsColumnName: String,
+      generalThreshold: Option[Double],
+      benchmarkThreshold: Map[String, Double])(
+      implicit L: Logger[F],
+      S: Sync[F],
+      C: Console[F],
+      M: Monad[F]): EitherT[F, HoodError, List[BenchmarkComparisonResult]] = {
+
+    val csvService = CsvService.build[F]
+
+    val parseFunction: File => F[Either[HoodError, List[Benchmark]]] = csvService.parseBenchmark(
+      BenchmarkColumns(
+        keyColumnName,
+        modeColumnName,
+        compareColumnName,
+        thresholdColumnName,
+        unitsColumnName))
+
+    for {
+      previousBenchmarks <- EitherT(parseFunction(previousPath)).map(buildBenchmarkMap)
+      currentBenchmarks  <- EitherT(parseFunction(currentPath)).map(buildBenchmarkMap)
+      result <- EitherT.right[HoodError](
+        performBenchmarkComparison[F](
+          currentBenchmarks,
+          previousBenchmarks,
+          generalThreshold,
+          benchmarkThreshold))
+      outputMessage = benchmarkOutput(result, previousPath.getName, currentPath.getName)
+      _ <- EitherT.right(C.putStrLn(outputMessage))
+    } yield result
+
+  }
+
+  private[this] def buildBenchmarkMap(benchmarks: List[Benchmark]): Map[String, Benchmark] =
+    benchmarks.map(b => (b.benchmark, b)).toMap
+
+  private[this] def performBenchmarkComparison[F[_]](
+      currentBenchmarks: Map[String, Benchmark],
+      previousBenchmarks: Map[String, Benchmark],
+      generalThreshold: Option[Double],
+      thresholdMap: Map[String, Double])(
+      implicit L: Logger[F],
+      S: Sync[F],
+      M: Monad[F]): F[List[BenchmarkComparisonResult]] =
+    previousBenchmarks
+      .map {
```

Comment:
I think you could swap this `map` by `traverse` and remove the `.sequence` at the end